### PR TITLE
Enable OpenSearch Security [DOC-241]

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
# What changed, and why it matters

OpenSearch Security content
 - About OS security, Access control in Aiven 
 - How to enable access control via console

https://aiven.atlassian.net/browse/DOC-241

### CODE REVIEW

File: docs/products/opensearch/howto/enable-opensearch-security.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+:doc:`OpenSearch Security <../concepts/os-security>` provides a range of security features, including fine-grained access controls, SAML authentication, and audit logging to monitor activity within your OpenSearch service. By enabling this, you can easily manage user permissions, roles, and other security aspects through the OpenSearch Dashboard.
+
+This article provides information on how to enable OpenSearch Security from the Aiven Console. 
+
+Considerations before enabling OpenSearch Security management
+-------------------------------------------------------------
+
+Before enabling OpenSearch Security management on your Aiven for OpenSearch service, please note the following important details:
+
+* OpenSearch Security management cannot be disabled once enabled. Therefore, ensure that you thoroughly understand the security features and implications before proceeding.
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/enable-opensearch-security.rst]: Fixed

File: docs/products/opensearch/concepts/os-security.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  .. note:: 
+    Aiven for OpenSearch provides basic SAML integration for the beta release, and certain features, such as logout support and request signing, is not be included.
+
+* **OpenSearch Dashboard multi-tenancy:** OpenSearch Security provides OpenSearch Dashboard multi-tenancy, which allows you to have different tenants on your OpenSearch Dashboard. 
+
+* **OpenSearch audit-logs:** Aiven for OpenSearch Security includes OpenSearch Audit-logs, which allow you to document a historical record of activity for compliance purposes and other business policy enforcement.
+
+
+
+OpenSearch Security management changes and impacts
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/concepts/os-security.rst:47]: Fixed

File: docs/products/opensearch/concepts/os-security.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+----------------------------------------------------
+Enabling OpenSearch Security management on your Aiven for OpenSearch service through the Aiven console triggers several changes:
+
+* All user authentication and access control will be managed through the OpenSearch security Dashboard or OpenSearch Security API. You have the ability to add or remove users from pre-generated roles, while some roles remain reserved for Aiven's internal use.
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/concepts/os-security.rst]: Fixed

File: docs/products/opensearch/concepts/access_control.rst [hunk 4]
```diff
@@ -1,127 +1,22 @@
...
+For more information about access control, patterns and permissions, see :doc:`Understanding access control in Aiven for OpenSearch® <../concepts/users-access-controls>`. 
 
-1. ``deny``: explicitly deny access
-#. ``admin``: unlimited access to the index
-#. ``readwrite``: full access to documents
-#. ``read``: allow only searching and retrieving documents
-#. ``write``: allow updating, adding, and deleting documents
+Method 2: Enable OpenSearch® Security management |beta|
+--------------------------------------------------------
+Another way to manage user accounts, access control, roles, and permissions for your Aiven for OpenSearch® service is by :doc:`enabling OpenSearch® Security management <../howto/enable-opensearch-security>`. This method lets you use the OpenSearch dashboard to manage all aspects of your service's security.
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/concepts/access_control.rst]: Fixed

File: docs/products/opensearch/concepts/os-security.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+* All service users defined at the time of enabling OS security will be classified as internal users of OpenSearch, with the attribute ``Aiven_managed: False`` assigned to most users. However, the ``avnadmin`` and ``os-sec-admin`` users will have the attribute ``Aiven_managed:true``.
+* The ``os-sec-admin`` user will be mapped to the pre-generated role ``aiven_security_admin_access``, which provides unrestricted access to the service, including the OpenSearch Security API and Dashboard. Other users are not mapped to this role.
+* As an ``os-sec-admin`` user, you can add or remove users from pre-generated roles, but some roles cannot be changed or deleted.
+* Aiven's internal users and their permissions/roles, which are necessary for managing and operating the service, are hidden and reserved from customer view and modification.
```
[the remark this answers is not in the document]
Comment (mteiste) [reply docs/products/opensearch/concepts/os-security.rst:61]: This is correct, they are hidden, but maybe remove this as a separate point and "include" it in the previous one (maybe "but some roles..." to " but users and roles essential for managing and operating the service cannot be modified".

**Review by mteiste — CHANGES_REQUESTED** [review]
I would hold off merging this yet. Most likely all the roles/definitions that mention aiven are going to be changed to something more generic (to cover the oem situation).

**Review by hoang-vo — APPROVED** [review]
Looking good now  👍

**Pull request merged**